cmd/controller/main.go

```diff
@@ -43,6 +47,20 @@ func main() {
 	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
 	cloudProvider := metrics.Decorate(awsCloudProvider)
 
+	client := op.Manager.GetAPIReader()
+	nodeClassList := v1beta1.EC2NodeClassList{}
+	err := client.List(ctx, &nodeClassList)
+	if err != nil {
+		logging.FromContext(ctx).Fatalf("failed to list EC2NodeClasses on startup, %s", err.Error())
+	}
+
+	invalidNodeClasses := lo.FilterMap(nodeClassList.Items, func(nc v1beta1.EC2NodeClass, _ int) (string, bool) {
+		return nc.Name, len(nc.Spec.AMISelectorTerms) == 0
+	})
+	if len(invalidNodeClasses) != 0 {
+		logging.FromContext(ctx).With("ec2nodeclasses", strings.Join(invalidNodeClasses, ",")).Fatalf("detected EC2NodeClasses with un-set AMISelectorTerms. Upgrade your EC2NodeClasses to include AMISelectorTerms to continue.")
```

Comment:
Should we make this error message more verbose about why this change was made and what's required to make the change here? From just reading this statement, it's not clear to me what I need to do e.g. "Upgrade your EC2NodeClasses". If I was just reading this, I would think that it just means that I need to bump the CRD version, which won't do anything here. Can we consider pointing to docs?

Reply:
I agree, this should probably point to the docs. I still need to make the update to the upgrade guide but I'll have it point there.